### ISSUE TITLE
`dotnet-coverage collect` returns successful exit code even when the `dotnet test` sub-process fails

### ISSUE THREAD

**Issue by djorgensendk** [microsoft/vstest#4094]
## Description

We use `dotnet-coverage collect` in our CI pipelines, with `dotnet test` running a test base as the subprocess.

In case any tests in the test base fails, `dotnet test` returns an exit code = 1. However, the exit code returned
by `dotnet-coverage collect` is 0.

## Steps to reproduce
The behavior can be verified on your local machine using Powershell

1. Create a C# solution with a project that must be tested, and a test project
1. Write a simple testable method such as:

    ``` csharp
    public int bar(int a, int b)
    {
        return a + b;
    }
    ```

1. Write a failing test such as:

    ``` csharp
    [Fact]
    public void Failure1()
    {
        var foo = new Foo();

        var result = foo.bar(1, 1);

        Assert.Equal(3, result);
    }
    ```

1. In Powershell:
   1. Restore and build the solution.
   1. Run `dotnet test <path to test-dll>` on the Test-dll created by building the solution.
   1. Verify using `$LASTEXITCODE` that the exit code was 1.
   1. Run `dotnet-coverage collect -f xml -o TestResults\coverage.xml dotnet test <path to test-dll`
   1. Verify using `$LASTEXITCODE` that the exit code was 0.

## Expected behavior
`dotnet-coverage collect` releays the exit code of the subproces, i.e. 1 instead 0 when dotnet test failed

## Actual behavior
The `dotnet-coverage collect` operation itself ran successfully, hence it returns exit code 0.

**Comment by DerGary** [comment]
Having the same problem right now, Is there something we can do?

**Comment by jakubch1** [comment]
Issue is fixed in version 17.5.0: https://www.nuget.org/packages/dotnet-coverage/17.5.0
